ZenWithTerms/tla/ZenWithTerms.tla

```diff
@@ -0,0 +1,301 @@
+-------------------------------------------------------------------------------------
+
+-------------------------------- MODULE ZenWithTerms --------------------------------
+\* Imported modules used in this specification
+EXTENDS Naturals, FiniteSets, Sequences, TLC
+
+----
+
+CONSTANTS Values
+
+\* Set of node ids (all master-eligible nodes)
+CONSTANTS Nodes
+
+\* RPC message types
+CONSTANTS
+  Join,
+  PublishRequest,
+  PublishResponse,
+  Commit
+
+----
+
+\* Set of requests and responses sent between nodes.
+VARIABLE messages
+\* Transitive closure of value updates as done by leaders 
+VARIABLE descendant
+
+\* node state (map from node id to state)
+VARIABLE currentTerm
+VARIABLE currentConfiguration \* committed config
+VARIABLE lastAcceptedTerm
+VARIABLE lastAcceptedVersion
+VARIABLE lastAcceptedValue
+VARIABLE lastAcceptedConfiguration
+VARIABLE joinVotes
+VARIABLE allowElection
+VARIABLE electionWon
+VARIABLE publishVersion
```

Comment:
I wonder whether `lastPublishedVersion` would be a better name, in parallel with `lastPublishedConfiguration`?